File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        lp = 0
        rp = len(nums) - 1

        if nums[lp] == target:
            return lp

        if nums[rp] == target:
            return rp

        while lp <= rp:
            midPointer = int((lp + rp)/2)
            print(lp, rp, midPointer)

            if nums[midPointer] == target:
                return midPointer
            elif nums[midPointer] < target:
                lp = midPointer + 1
                continue
            else:
                rp = midPointer - 1
                continue
        
        return -1

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> intCounter;

        // 1. Insert all numbers in map by occurence
        for(const auto& num : nums){
            intCounter[num]++;
        }

        // 2. Priority queue to sort map values in descending order
        priority_queue<pair<int,int>> pq;
        for(const auto& pair : intCounter){
            pq.push({pair.second,pair.first});
        }

        // 3. Insert pq in result-vector
        vector<int> result;
        for(int i = 0; i < k; ++i){
            result.push_back(pq.top().second);
            pq.pop();
        }

        return result;
    }
};
